app/helpers/order_cycles_helper.rb

```diff
@@ -36,6 +36,15 @@ def permitted_hub_enterprise_options_for(order_cycle)
                                  shipping_and_payment_methods: true
   end
 
+  def distributors_with_editable_shipping_and_payment_methods(order_cycle)
+    return order_cycle.distributors if order_cycle.coordinator
+      .in? Enterprise.managed_by(spree_current_user)
+
+    order_cycle.distributors.select do |distributor|
+      distributor.in? Enterprise.managed_by(spree_current_user)
+    end
```

Comment:
I would hope that `Enterprise.managed_by(spree_current_user)` is cached but it wouldn't hurt to store it in a variable first. It could also be represented as a union:
```suggestion
    order_cycle.distributors & Enterprise.managed_by(spree_current_user)
```

But super-admin can see all enterprises which would make this list very long. So we could add a condition to shortcut super admin. 

Oh, hang on, I'm just realising that we completely missed the greatest feature of ActiveRecord scopes: chaining!

```suggestion
    order_cycle.distributors.managed_by(spree_current_user)
```

I reckon that this change is worth doing before merging.

Reply:
Okay, tried that and it turns out that `order_cycle.distributors` is an array, not a scope. Oh, hang on, just in the spec...